### PR TITLE
Easi 3621 funding source updates

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
# EASI-3621

## Changes and Description

- Backend
    - Add four new columns to describe medicare a and b for the when selected for funding source or funding source r
    - Update GQL
    - Create Migration to handle historic data
- Front End
  - Update translation with new values for `fundingSource`
  - Add additional translation for new `other`  fields
  - Added standardization for `Other no specified` and `No additional information specified` fields in Readonly
  - Updated Filter Group mappings for new fields
  - Updated `ReadonlySection` component to handle multiple fields with optional additional info
    - Left legacy prop for only one field, decided to make a new prop to handle an array of fields
    - Added a prop to handle rendering `Tooltip` in readonly in addition to form components
  - Render new form fields
  - Update Readonly to render new fields
  - Unit tests / Cypress tests

## How to test this change

1. Bring up the app
2. Navigate to plan payments
3. Answer all the new questions

To test the migration, 
1. Comment out the new migration
2. Bring up the app (running all the other migrations)
3. Run test data  SQL to get some test data
4. Run the final migration ( copy and paste it in your chosen SQL tool)
5. Make sure to clean the database when you are done with this test, because the test data doesn't have all of the task list sections. (So some queries will fail)

test data SQL
```sql
INSERT INTO "public"."user_account"("id", "username", "is_euaid", "common_name", "locale", "email", "given_name", "family_name", "zone_info", "has_logged_in") VALUES('48523a41-08e0-4a21-b2b9-b0cacd9615ff', 'MINT', 'TRUE', 'MINT Doe', 'en_US', 'MINT@local.cms.gov', 'MINT', 'Doe', 'America/Los_Angeles', 'TRUE');

-- Option with only trust fund selected for funding_source testing
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d21', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source", "created_dts", "modified_dts", "status", "created_by", "modified_by") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf0', '8780b574-4a2b-4485-9537-c76813c57d21', '{TRUST_FUND}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
UPDATE "public"."plan_payments" SET "funding_source"='{TRUST_FUND,OTHER}', "funding_source_other"='(existing text)' WHERE "id"='ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf0' ;

-- Option with only trust fund selected for funding_source_r testing
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d22', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source_r", "created_dts", "modified_dts", "status", "created_by", "modified_by") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf1', '8780b574-4a2b-4485-9537-c76813c57d22', '{TRUST_FUND}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');

-- Option with only trust fund selected for funding_source and funding_source_r testing
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d23', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source_r","funding_source", "created_dts", "modified_dts", "status", "created_by", "modified_by") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf2', '8780b574-4a2b-4485-9537-c76813c57d23', '{TRUST_FUND}', '{TRUST_FUND}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');


-- Option with trust fund selected for funding_source and funding_source_r testing. Both medicare types selected for both fields
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d24', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source", "funding_source_r", "created_dts", "modified_dts", "status", "created_by", "modified_by", "funding_source_trust_fund_type", "funding_source_r_trust_fund_type") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf3', '8780b574-4a2b-4485-9537-c76813c57d24', '{TRUST_FUND}', '{TRUST_FUND}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}');

-- Option with trust fund selected, then removed for funding_source and funding_source_r testing. Both medicare types selected for both fields
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d25', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source", "funding_source_r", "created_dts", "modified_dts", "status", "created_by", "modified_by", "funding_source_trust_fund_type", "funding_source_r_trust_fund_type") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf4', '8780b574-4a2b-4485-9537-c76813c57d25', '{}', '{}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}');


-- Option with trust fund and PATIENT_PROTECTION_AFFORDABLE_CARE_ACT selected for funding_source and funding_source_r testing. Both medicare types selected for both fields
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d26', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source", "funding_source_r", "created_dts", "modified_dts", "status", "created_by", "modified_by", "funding_source_trust_fund_type", "funding_source_r_trust_fund_type") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf5', '8780b574-4a2b-4485-9537-c76813c57d26', '{TRUST_FUND,PATIENT_PROTECTION_AFFORDABLE_CARE_ACT}', '{TRUST_FUND, PATIENT_PROTECTION_AFFORDABLE_CARE_ACT}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}', '{MEDICARE_PART_A_HI_TRUST_FUND,MEDICARE_PART_B_SMI_TRUST_FUND}');


-- Option with trust fund selected for funding_source and funding_source_r testing. Trust fund is an empty array, but not null
INSERT INTO "public"."model_plan"("id", "model_name", "archived", "status", "created_dts", "created_by") VALUES('8780b574-4a2b-4485-9537-c76813c57d27', 'Excellent Model', 'FALSE', 'PLAN_DRAFT', '2023-12-07 19:21:37.759997+00', '48523a41-08e0-4a21-b2b9-b0cacd9615ff');
INSERT INTO "public"."plan_payments"("id", "model_plan_id", "funding_source", "funding_source_r", "created_dts", "modified_dts", "status", "created_by", "modified_by", "funding_source_trust_fund_type", "funding_source_r_trust_fund_type") VALUES('ffe4b9ab-e9e1-48be-b5f2-eeac5c9eacf6', '8780b574-4a2b-4485-9537-c76813c57d27', '{TRUST_FUND}', '{TRUST_FUND}', '2023-12-07 19:21:37.759997+00', '2023-12-07 19:21:37.759997+00', 'READY', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '48523a41-08e0-4a21-b2b9-b0cacd9615ff', '{}', '{}');

```

Frontend: 

- Navigate to the first page of `Payments`
- Click on `Medicare Part A (HI) Trust Fund` and/or `Medicare Part B (SMI) Trust Fund`
- Verify the conditional text field render on selection
- Verify the persistence of data on form reentry
- Verify the data renders in the readonly
- Verify the optional text, if empty, renders the correct text for other/no additional info
- Verify the ToolTips render in readonly
- Verify the data exists in csv export

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/models/plan_payments.go [hunk 1]
```diff
@@ -129,26 +133,21 @@ const (
...
-	FundingSource                   pq.StringArray `json:"fundingSource" db:"funding_source" statusWeight:"1"`
-	FundingSourceTrustFundType      pq.StringArray `json:"fundingSourceTrustFundType" db:"funding_source_trust_fund_type"`
-	FundingSourceOther              *string        `json:"fundingSourceOther" db:"funding_source_other"`
-	FundingSourceNote               *string        `json:"fundingSourceNote" db:"funding_source_note"`
-	FundingSourceR                  pq.StringArray `json:"fundingSourceR" db:"funding_source_r" statusWeight:"1"`
-	FundingSourceRTrustFundType     pq.StringArray `json:"fundingSourceRTrustFundType" db:"funding_source_r_trust_fund_type"`
+	FundingSource              pq.StringArray `json:"fundingSource" db:"funding_source" statusWeight:"1"`
+	FundingSourceMedicareAInfo *string        `json:"fundingSourceMedicareAInfo" db:"funding_source_medicare_a_info"`
+	FundingSourceMedicareBInfo *string        `json:"fundingSourceMedicareBInfo" db:"funding_source_medicare_b_info"`
+	// FundingSourceTrustFundType      pq.StringArray `json:"fundingSourceTrustFundType" db:"funding_source_trust_fund_type"`
```
Comment (OddTomBrooks) [inline pkg/models/plan_payments.go:144]: Is there a reason we're keeping this?

File: pkg/models/plan_payments.go [hunk 2]
```diff
@@ -129,26 +133,21 @@ const (
...
+	FundingSourceOther          *string        `json:"fundingSourceOther" db:"funding_source_other"`
+	FundingSourceNote           *string        `json:"fundingSourceNote" db:"funding_source_note"`
+	FundingSourceR              pq.StringArray `json:"fundingSourceR" db:"funding_source_r" statusWeight:"1"`
+	FundingSourceRMedicareAInfo *string        `json:"fundingSourceRMedicareAInfo" db:"funding_source_r_medicare_a_info"`
+	FundingSourceRMedicareBInfo *string        `json:"fundingSourceRMedicareBInfo" db:"funding_source_r_medicare_b_info"`
+	// FundingSourceRTrustFundType     pq.StringArray `json:"fundingSourceRTrustFundType" db:"funding_source_r_trust_fund_type"`
```
Comment (OddTomBrooks) [inline pkg/models/plan_payments.go:150]: Is there a reason we're keeping this?

**Comment by StevenWadeOddball** [comment]
Thanks for your feedback @OddTomBrooks . I addressed your comments an implemented the changes. Anything else for the backend?

File: pkg/models/plan_payments.go (hunk 2 above)
Comment (StevenWadeOddball) [reply pkg/models/plan_payments.go]: Good catch!

File: pkg/models/plan_payments.go (hunk 1 above)
Comment (StevenWadeOddball) [reply pkg/models/plan_payments.go]: Good catch!

File: migrations/V121__Update_Funding_Source_Type.sql [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ALTER TABLE plan_payments
+    ADD COLUMN funding_source_medicare_a_info ZERO_STRING,
+    ADD COLUMN funding_source_medicare_b_info ZERO_STRING,
+    ADD COLUMN funding_source_r_medicare_a_info ZERO_STRING,
+    ADD COLUMN funding_source_r_medicare_b_info ZERO_STRING;
+
+ALTER TABLE plan_payments
+ALTER COLUMN funding_source TYPE TEXT[];
+ALTER TABLE plan_payments
+ALTER COLUMN funding_source_r TYPE TEXT[];
```
[the remark this answers is not in the document]
Comment (StevenWadeOddball) [reply migrations/V121__Update_Funding_Source_Type.sql]: good catch! That would work as well. I implemented it

**Review by OddTomBrooks — APPROVED** [review]
LGTM. All indications are good. Let's keep a close eye on this one due to the complexity.

**Review by garyjzhao — APPROVED** [review]
FE LGTM.
Data gets recorded ✅
Read Only ✅
CSV ✅

**Pull request merged**